lib/script-generator.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+var FsDriver = require('./fs-driver');
+
+/**
+ * Generates shell scripts from transformation rules.
+ * @module fs-transform:script
+ * @author Ryan Sandor Richards
+ */
+module.exports = ScriptGenerator;
+
+/**
+ * Creates a new ScriptGenerator class that can generate scripts for the given
+ * set of transformation rules. Note: rules are not validated before generating
+ * the script. Please ensure you validate all rules using `Transformer.dry`
+ * before generating shell scripts.
+ * @class
+ * @param {array} rules Rules to convert into a shell script.
+ */
+function ScriptGenerator(rules) {
+  this.setRules(rules);
+  this.actionHandlers = {
+    'copy': this.copy.bind(this),
+    'rename': this.rename.bind(this),
+    'replace': this.replace.bind(this),
+    'exclude': this.exclude.bind(this)
+  };
```

Comment:
Technically no, they don't need to be bound. However, since we are calling methods on an instance I'd like to have the methods `this` be correct from the programmer's point of view.